configs/defs/elasticsearch.textproto

```diff
@@ -0,0 +1,733 @@
+        "  }"
+        "}"
+  
+        # /etc/elasticsearch/elasticsearch.yml
+        "check_alternatives:{"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/etc/elasticsearch/elasticsearch.yml\"}}"
+        "    existence:{should_exist: true}"
+        "    non_compliance_msg: \"Elasticsearch configuration file does not exist\""
+        "  }"
+        "  file_checks:{"
+        "    files_to_check:{files_in_dir:{"
+        "      dir_path:\"/etc/elasticsearch\""
+        "      recursive: true"
+        "      files_only: true"
+        "    }}"
+        "    permission:{ user: {name: \"elasticsearch\" should_own: true} }"
+        "  }"
+        "}}"
+    }
+  }
+  
+  benchmark_configs: {
+    id: "elasticsearch-configuration-permissions"
+    compliance_note: {
+      version: { version: "1.0.0" benchmark_document: "" }
+      title: "Ensure ElasticSearch Configuration has Correct Permissions"
+      description:
+        "This check verifies that the Elasticsearch configuration file has the correct permissions. "
+        "The configuration file contains sensitive information, including connection details, "
+        "security settings, and other important parameters, and it is crucial to "
+        "protect it from unauthorized access and modifications."
+      rationale:
+        "Setting the correct permissions for the Elasticsearch configuration file is essential for "
+        "maintaining the security and integrity of the Elasticsearch deployment. By ensuring that "
+        "the file has the correct permissions set, it is possible to restrict access to authorized "
+        "users or processes only. This prevents unauthorized users from viewing or modifying critical "
+        "configuration settings, reducing the risk of misconfigurations or security breaches."
+      remediation:
+        "To remediate this issue, follow these steps:\n"
+        "1. Identify the location of the Elasticsearch configuration file in your environment.\n"
+        "2. Run the following command to set the correct permissions for the configuration file:\n"
+        "      chmod 640 <Path_to_Config_File>\n"
+        "   Replace <Path_to_Config_File> with the actual path to the Elasticsearch configuration file.\n"
+        "3. Verify that the permissions have been set correctly by running the following command:\n"
+        "      ls -l <Path_to_Config_File>\n"
+        "   The command should display the permissions as `-rw-r-----` for the Elasticsearch configuration file."
+      cis_benchmark: {
+        profile_level: 1
+        severity: LOW
+      }
+      scan_instructions:
+        # /usr/share/elasticsearch/config/elasticsearch.yml
+        "generic{check_alternatives:{"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/usr/share/elasticsearch/config/elasticsearch.yml\"}}"
+        "    existence:{should_exist: true}"
```

Comment:
Same here, I think the permisison check fails anyways if the file doesn't exist.